### DISCUSSION TITLE
Coregistered template doesn't always find the head

### DISCUSSION THREAD

**Discussion by Kaiser-Neuroscan** [simnibs/simnibs#355] [7.Creating head models] [answered]
![image](https://github.com/simnibs/simnibs/assets/171934171/dca548a1-c1d9-4eb6-ad31-71a415990010)
Hi everyone!
I have tried to do some segmentations with CHARM and I have run into an issue, where the coregistered template doesn't match the head at all and the result of the segmentation is distorted and probably made up since in that region there isn't much to segment.
Has anyone encountered this problem and knows how to fix it?
Thanks!

**Comment by oulap** [comment]
Hey,

Is this possibly a younger subject? It seems there's a lot of neck at least, which can sometimes cause problems. If you look into the m2m-folder of that subject you should see a file called settings.ini. Please copy that and place it somewhere outside the m2m-folder. You can then edit these fields in the settings.ini file to initialize the affine a bit differently:

```
[initatlas]
# Scaling factors to search through as an initialization default: [0.85, 0.9, 0.95]
affine_scales = [[0.85, 0.85, 0.85], [0.9, 0.9, 0.85], [0.95, 0.95, 0.85]]
# Rotations (around LR axis, degrees) to search through default: [-7, -3.5, 0, 3.5, 7]
affine_rotations = [-7, -3.5, 0, 3.5, 7]
# Horizontal shifts (in mm) to search through default: [-20.0, -10.0, 0, 10.0, 20.0]
affine_horizontal_shifts = [-20.0, -10.0, 0, 10.0, 20.0]
# Vertical shifts to search through (in mm) default: [-10.0, 0.0, 10.0]
affine_vertical_shifts = [-10.0, 0.0, 10.0]
# Neck deformation bounds default: [-0.4, 0.1]
neck_search_bounds = [-0.3, 0.1]
# Downsampling target resolution (in mm) default: 2
downsampling_factor_affine = 2
```

For your case I think changing the vertical shift makes most sense. Maybe you can just first try to fix it to 

`affine_vertical_shifts = [10.0]`

and see what happens. To use the new, edited settings file please add this flag to the end of your charm command:

`--usesettings <path_to_settings>`

where `<path_to_settings>` is the path to the edited settings file.

**Reply by Kaiser-Neuroscan** [reply]
Thanks for the advice! Unfortunately, the issue still arises:
![image](https://github.com/simnibs/simnibs/assets/171934171/3fbfee6d-a7c5-4354-845e-9c4b4bd9207c)

What you say sounds like cutting off the neck would help, should I try that next?

**Reply by oulap** [reply, marked as answer]
Can you try one run with the vertical shift set to:

`affine_vertical_shifts = [10.0]`

and another with the vertical shift set to:

`affine_vertical_shifts = [-10.0]`

and see if one of those gives better results. Please do **not** include zero here, only either 10 or -10. I can see in the picture you posted that the shifts are set to:

`affine_vertical_shifts = [10.0, 0.0 , 10.0]`

that might still end up using no shift, i.e., 0.0. You can also play with the scaling as it looks to me that the head in the MRI scan might be a bit smaller than the average head we have in the atlas.

**Comment by Kaiser-Neuroscan** [comment]
Hi Oula,

a vertical shift of 40 worked! As you noticed, the coregistered template is a bit larger than the head and even though the segmentation looks quite good, the skin compartment can have some protruding "lumps". Should be fine though.
Thank you so much!
![image](https://github.com/simnibs/simnibs/assets/171934171/bc9c4ca5-a7c1-485b-9080-7b7dee80f1a7)
![image](https://github.com/simnibs/simnibs/assets/171934171/79f1d8bb-c428-4d72-bc33-8b762aa4c035)

**Reply by oulap** [reply]
Great! You can maybe get of the skin lumps if you also edit the affine_scales field, e.g., something like this might work:

`affine_scales = [[0.85, 0.85, 0.85]]`

That should shrink the template initially (it looks like it's still a bit too big in the screenshot above).
